File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int cur = INT_MIN, ans = 0;
        
        for(int p : prices){            
            int a = cur, b = ans;
            cur = max(a, b - p);
            ans = max(b, a + p);
        }
        
        return ans;
    }
};

File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        cur = -0x3f3f3f3f
        ans = 0
        
        for p in prices:
            a, b = cur, ans
            cur = max(a, b-p)
            ans = max(b, a+p)
            
        return ans;